knime_extension/src/nodes/spatialnetwork.py

```diff
@@ -754,7 +764,7 @@ def ckdnearest(gdA, gdB):
         O_gdf = gp.GeoDataFrame(O_gdf, geometry="geometry")
         D_gdf = gp.GeoDataFrame(D_gdf, geometry="geometry")
         R_gdf = gp.GeoDataFrame(R_gdf, geometry="geometry")
-        R_gdf = R_gdf.rename(columns={self.Speed_col: "speed"})
+        R_gdf = R_gdf.rename(columns={self.R_speed_col: "speed"})
```

Comment:
Why do you rename all of these columns instead of using the knut.load_geo_data_frame() method which takes care of assigning the selected column as geometry column or does the column name need to be geometry?
What happens if one of these dataframes already contains a column named geometry?

Reply:
as the code only needs geometry and speed info, I revise the code as follows:
        # Cross join Data
        O_gdf = knut.load_geo_data_frame(input1, self.O_geo_col, exec_context)
        D_gdf = knut.load_geo_data_frame(input2, self.D_geo_col, exec_context)
        R_gdf = knut.load_geo_data_frame(input3, self.R_geo_col, exec_context)
        O_gdf = O_gdf[self.O_geo_col].rename(columns={self.O_geo_col: "geometry"})
        D_gdf = D_gdf[self.D_geo_col].rename(columns={self.D_geo_col: "geometry"})
        R_gdf = R_gdf[self.R_geo_col,self.R_speed_col].rename(columns={self.R_geo_col: "geometry",self.R_speed_col: "speed"})